spec/io/sql_data_source_spec.rb

```diff
@@ -71,13 +43,28 @@
       it { expect(df.row[0][:name]).to eq 'Homer' }
     end
 
-    context 'with path to unsupported db file' do
+    context 'with an object not a string as a query' do
       it {
         expect {
-          Daru::IO::SqlDataSource.make_dataframe("spec/fixtures/bank2.dat", query)
+          Daru::IO::SqlDataSource.make_dataframe(active_record_connection, Object.new)
         }.to raise_error(ArgumentError)
       }
     end
 
+    context 'with an object not a database connection' do
+      it {
+        expect {
+          Daru::IO::SqlDataSource.make_dataframe(Object.new, query)
+        }.to raise_error(ArgumentError)
```

Comment:
If you'd follow my suggestion about common subject, this context could be turned into
```ruby
let(:source) { Object.new }
it { expect { subject }.to raise_error(ArgumentError) }
```

And I'd suggest to check error message too.